### ISSUE TITLE
Optimized WAR have trouble in using TTFFONT while generating PDF using UI-GRID

### ISSUE THREAD

**Issue by saravp2016** [bpampuch/pdfmake#835]
Hi,

I am using UI-GRID for my reporting which in turn uses PDFMAKE for generating PDF export. It's a Jhipster generated app(uses Spring Boot+Angular JS). 

Everything works fine in my Dev environment. But PDF generation is not happening only in my Prod environment. Jhipster have Prod profile which will generate optimized WAR with compressed/uglified js. Am not sure where the problem lies, still debugging.

**Findings so far:**
As you can refer in below screenshot, the TTFFONT referred is 'Roboto-Italic.ttf', the file is read(I believe from vfs_fonts.js), a buffer stream is passed, and from there a directory is decoded and below font is generated.

While my local font have 'Name' object filled, my prod font's 'Name' object is undefined :(. 
So when trying to retrieve 'postScriptName' through this.name.records.postScriptName, it throws error like 'records' of undefined property, something like that. 

All above things happen in pdfmake.js, so raised issue here. Any help why this is happening? I have vfs_fonts.js also loaded, seems no issues with that as it decodes most of the other directories except for 'Name' object,
So any idea or solution on how to fix this up? Do I need to do something with the font? Thanks in advance.

Local Environment:
![image](https://cloud.githubusercontent.com/assets/20683750/22141680/3149daf6-df19-11e6-9d8b-b133e683968f.png)

Prod Environment

![image](https://cloud.githubusercontent.com/assets/20683750/22141423/093290e0-df18-11e6-9e7c-4e73d17a9260.png)

**Comment by liborm85** [comment]
There was a similar problem https://github.com/bpampuch/pdfmake/issues/828 (problem was in HTML codepage). What are your codepage? UTF-8 required.

**Comment by saravp2016** [comment]
Sorry for late response, I was away during weekend.

@liborm85 Thanks for referring to the existing issue, I believe here also it should be HTML encoding related issue.

But I got below HTML tags already in my HTML. Still I face this issue in my production environment.
_<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">_

In my local Dev enviornment I use Spring boot's embedded tomcat server, while in my Prod I use Jboss server. Since the HTML code is same and also it has necessary tags with it, I doubt it could be something to do with server.

So looking into my Jboss server to have any configuration to force UTF-8 encoding for the HTML served. Not sure this is right approach, I appreciate if somebody can validate this approach or provide a solution in this regard.

**Comment by saravp2016** [comment]
The issue was fixed. Root cause was outdated HTML encoding.

Jhipster have implemented EmbeddedServletContainerCustomizer in their WebConfiguer.java to add charset(UTF-8) information for their HTML pages served through embedded tomcat server.

But external server(in my case jboss) by default was considering some outdated encoding unless we force it to serve UTF-8 encoded HTML pages. Since it was a spring boot application, I set the charset details in their application.yml/properties file as below and it worked:
![image](https://cloud.githubusercontent.com/assets/20683750/22245404/b53fec08-e255-11e6-8eb4-dd0a4813eefe.png)

Thanks @liborm85  once again for guiding through right path. Since prod WAR was minified, uglified, etc I was considering those process would have broken the JS somewhere and spend more time on it. I would have never thought of HTML encoding as cause of the issue.

Am closing the issue as it was fixed.
